pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -666,14 +685,29 @@ func (p *PriorityQueue) isPodBackingoff(podInfo *framework.QueuedPodInfo) bool {
 
 // SchedulingCycle returns current scheduling cycle.
 func (p *PriorityQueue) SchedulingCycle() int64 {
-	p.lock.RLock()
-	defer p.lock.RUnlock()
+	p.activeQLock.RLock()
+	defer p.activeQLock.RUnlock()
 	return p.schedulingCycle
 }
 
-// determineSchedulingHintForInFlightPod looks at the unschedulable plugins of the given Pod
-// and determines the scheduling hint for this Pod while checking the events that happened during in-flight.
-func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger, pInfo *framework.QueuedPodInfo) queueingStrategy {
+// dumpClusterEvents dumps all cluster events that have happened during this inFlightPod is being scheduled.
+// Note: this function assumes activeQLock to be locked by the caller.
+func (p *PriorityQueue) dumpClusterEvents(inFlightPod *list.Element) []*clusterEvent {
```

Comment:
```suggestion
func (p *PriorityQueue) clusterEventsSinceElementUnlocked(inFlightPod *list.Element) []*clusterEvent {
```

We usually use "dump" when implementing some debugging function, so I would avoid that word.

Reply:
Good to know